openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -263,7 +263,25 @@ def get_libraries_for_user(user, org=None, library_type=None, text_search=None):
             Q(learning_package__description__icontains=text_search)
         )
 
-    return permissions.perms[permissions.CAN_VIEW_THIS_CONTENT_LIBRARY].filter(user, qs)
+    filtered = permissions.perms[permissions.CAN_VIEW_THIS_CONTENT_LIBRARY].filter(user, qs)
+
+    if order:
+        order_query = 'learning_package__'
+        if '-' in order:
+            order_query = f"-{order_query}"
+            order = order.replace("-", "")
```

Comment:
nit: this is just about the asc/desc flag, isn't it? I think that should be more explicit:

```suggestion
        # If order starts with a -, that means order descending (default is ascending)
        if order.startswith('-'):
            order_query = f"-{order_query}"
            order = order[1:]
```

Reply:
Much cleaner! I've made the change [cfb9f55](https://github.com/openedx/edx-platform/pull/35005/commits/cfb9f558282095bce7f71e1cff13dad9432a6323)